articles/quickstart/native/ionic-angular/01-login.md

```diff
@@ -0,0 +1,245 @@
+- `cacheLocation`: The location at which to store tokens. We use `localstorage` here so that tokens are persisted across app refreshes.
+
+<%= include('../_includes/ionic/_note_storage') %>
+
+:::panel Checkpoint
+Now that you have configured your app with the Auth0 Angular SDK, run your application to verify that the SDK is initializing without error and that your application runs as it did before.
+:::
+
+## Add Login to Your Application
+
+<%= include('../_includes/ionic/_add_login_intro') %>
+
+Add a new `LoginButton` component to your application with the following code:
+
+```js
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '@auth0/auth0-angular';
+import { Browser } from '@capacitor/browser';
+import { tap } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-login-button',
+  template: `<ion-button (click)="login()">Login</ion-button>`,
+})
+export class LoginButtonComponent {
+  constructor(public auth: AuthService) {}
+
+login() {
+    this.auth
+      .buildAuthorizeUrl()
+      .pipe(tap((url) => Browser.open({ url, windowName: '_self' })))
+      .subscribe();
+  }
+}
+```
+
+This component:
+
+* defines a template with a simple button that logs the user in when clicked
+* uses `buildAuthorizeUrl` to construct a URL to Auth0's Universal Login page
+* uses Capacitor's Browser plugin to open the URL and show the login page to the user
+
+### Handling the callback
+
+<%= include('../_includes/ionic/_handle_callback_intro') %>
+
+Modify your `App` component and use the `ngOnInit` method to handle the callback from Auth0:
+
+```js
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from '@auth0/auth0-angular';
+import { tap } from 'rxjs/operators';
+import { Browser } from '@capacitor/browser';
+import { App } from '@capacitor/app';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: 'app.component.html',
+  styleUrls: ['app.component.scss'],
+})
+export class AppComponent implements OnInit {
+  // Import the AuthService module from the Auth0 Angular SDK
+  constructor(public auth: AuthService) {}
+
+  ngOnInit(): void {
+    // Use Capacitor's App plugin to subscribe to the `appUrlOpen` event
+    App.addListener('appUrlOpen', ({ url }) => {
+      if (url) {
+        // If the URL is an authentication callback URL..
+        if (
+          url.includes('state=') &&
+          (url.includes('error=') || url.includes('code='))
+        ) {
+          // Call handleRedirectCallback and close the browser
+          this.auth
+            .handleRedirectCallback(url)
+            .pipe(tap(() => Browser.close()))
+            .subscribe();
+        }
+
+        Browser.close();
```

Comment:
Isnt this calling `Browser.close()` instantly when there is a state and code/error param?

Reply:
I've modified this as discussed - it now checks for the full callback URI on line 142, and the `Browser.close()` is now in an `else` block if the parameters aren't there.